### ISSUE TITLE
MQTT Error when starting the Webview

### ISSUE THREAD

**Issue by andyinno** [thanksmister/wallpanel-android#93]
**Describe the bug**
I just installed Wallpanel on a Nexus 5.
I configured MQTT to connect to Home assistant (running on Hass)
When connecting the device display the message: "There was an error initializing the MQTT client, please check your settings."

I configured the user on Home assistant and in MQTT add-on I see:

    New connection from 192.168.2.240 on port 1883.
    [INFO] found **redacted** on Home Assistant
    Socket error on client <unknown>, disconnecting.

**To Reproduce**
1. Configure MQTT client username and password
2. Open DASHBOARD
3. Wait for the web page to appear
4. The message is displayed in front of the web page

**Expected behavior**
To be able to communicate via MQTT with home assistant

**Smartphone (please complete the following information):**
 - Device: Nexus 5 and OnePlus 3T
 - OS: Android 6.0.1 on Nexus 5, Android 9 on OnePlus 3T
 - Version v8.0.7 Build 2

**Additional context**
I am quite sure that the settings are corrected. I entered them multiple times.
I just don't have any idea about how to debug the connection. It seems to me that the connection is closed before the client can completly connect. <unknown> should not be there, I added the client ID in the MQTT settings

**Comment by thanksmister** [comment]
I am seeing an increase suddenly in the amount of MQTT errors and crashes on a range of devices.  I am investigating the issue and trying some fixes.

**Comment by andyinno** [comment]
Thank you for your fast response,
if I can do something (testing, debugging) I am open to try new software or software with some kind of log in it.

**Comment by thanksmister** [comment]
You can test the version I just pushed out, I added more error reporting so I can see any crashes on the MQTT connections.   I am also bubbling up specific messages for different parts of the MQTT setup.

**Comment by andyinno** [comment]
Hello, thank you again.
I installed build3.

I don't see any difference with the previous version. There is something I have to do for see the error reporting?

**Comment by thanksmister** [comment]
No, the error reports come back to me in Firebase.  The only difference you might see is the popup alert message in the app when an error occurs.

**Comment by andyinno** [comment]
Ok. Understood.

I changed my setup with the information as displayed here: https://www.home-assistant.io/addons/mosquitto/

I updated the login and password in the app.
this is what I get:
    1560630601: New connection from 192.168.2.240 on port 1883.
    [INFO] found local-user on local database
    1560630603: Socket error on client <unknown>, disconnecting.

My test was just for make sure it is not hassio auth related.
I added directly the user to the setup.

**Comment by andyinno** [comment]
I don't know if it is relevant. I see two logins one after the other:
 
    1560630598: New connection from 192.168.2.240 on port 1883.
    [INFO] found local-user on local database
    1560630601: Socket error on client <unknown>, disconnecting.
    1560630601: New connection from 192.168.2.240 on port 1883.
    [INFO] found local-user on local database
    1560630603: Socket error on client <unknown>, disconnecting.

**Comment by thanksmister** [comment]
I don't see any new bug reports for build 3 or for  Nexus 5 device. I assume this issue is either your MQTT broker in HA or some issue with the MQTT settings in the app.  The HA errors don't tell me much other than its disconnecting clients, which could be something either wrong with your MQTT installation or something with the connect.  

This is what my settings look like for my MQTT setup in the app:

<img width="504" alt="mqtt" src="https://user-images.githubusercontent.com/142340/59556109-95574600-8f93-11e9-93f7-79d66c989466.png">

Do you have any specific errors displayed inside the app?

**Comment by andyinno** [comment]
No, just the error message.

![Screenshot_20190615-223658](https://user-images.githubusercontent.com/5983066/59556182-70c49380-8fbe-11e9-95d9-09dfea9047b3.png)

I will try to start another mosquitto server maybe on my laptop and test if it connects there.
Thank you for now.

**Comment by thanksmister** [comment]
In my MQTT server, I have the message for the wallpanel connected

```1560631227: Socket error on client mywallpanel, disconnecting.
1560631227: New client connected from 192.168.1.28 as mywallpanel (c0, k60, u'mister').```

**Comment by thanksmister** [comment]
That error message you see happened when your application tried to send a message, so it connected and was disconnected before it could send a message out, this probably happened after your socket connection to the MQTT broker was cut.  So this syncs with your MQTT broker errors.    I am not sure this is an app issue, it seems that your MQTT broker is disconnecting clients. 

This part of your MQTT error `Socket error on client <unknown>, disconnecting` . with the unknown device seems strange, did you change the client ID within your MQTT settings of the app?  It looks like its blank, but each client must have a client Id that is unique for the device.

**Comment by andyinno** [comment]
Ok, I found it.
It was my mistake.

Just as an explanation about what was happening and if someone else makes the same mistake:

I setup a mosquitto server on my laptop as anticipated and the app could connect without any issue.
I went back to the add-on setup page and verified the documentation.

https://www.home-assistant.io/addons/mosquitto/ from here the warning related to the ACL: https://www.home-assistant.io/addons/mosquitto/#access-control-lists-acls

it cames out that I mispelled the name in the acl file.
the copy & paste configuration was made without adding anything to the acl file

`
{
  "logins": [
    {"username": "local-user", "password": "mypw"}
  ],
  "anonymous": false,
  "customize": {
    "active": false,
    "folder": "mosquitto"
  },
  "certfile": "fullchain.pem",
  "keyfile": "privkey.pem"
}
`

So basically the same kind of error happened for both setups I tried.

1. The client was connecting to the server
2. The client was trying to write something
3. The write was rejected because the ACL file did not match the rule
4. The server disconnects the client after this error.

Thank you for your promt response, sorry for my mistake but it was all day long that I was trying to configure the client and this error was continuosly popping up without a clear error message.
